### PR TITLE
feature: added response types for only ids and added tag_set to update_chunk

### PR DESCRIPTION
**Pull request by densumesh**
## Please indicate what issue this PR is related to and @ any maintainers who are relevant
image
image

### CODE REVIEW

File: server/src/handlers/chunk_handler.rs [hunk 1]
```diff
@@ -1096,7 +1100,11 @@ pub fn parse_query(query: String) -> ParsedQuery {
     tag = "chunk",
     request_body(content = SearchChunkData, description = "JSON request payload to semantically search for chunks (chunks)", content_type = "application/json"),
     responses(
-        (status = 200, description = "chunks which are similar to the embedding vector of the search query", body = SearchChunkQueryResponseBody),
+        (status = 200, content(
+            ("application/json" = SearchChunkQueryResponseBody),
+            ("application/json.only_ids" = SearchChunkQueryIDsResponseBody)
```
Comment (fastfists) [inline server/src/handlers/chunk_handler.rs:1105]: How does this look in redoc?

File: server/src/data/models.rs [hunk 2]
```diff
@@ -439,7 +439,7 @@ impl From<ChunkMetadataWithFileData> for SlimChunkMetadata {
     }
 }
 
-#[derive(Debug, Serialize, Deserialize, Clone, Queryable)]
+#[derive(Debug, Serialize, Deserialize, Clone, Queryable, ToSchema)]
```
Comment (skeptrunedev) [inline server/src/data/models.rs:442]: Line below, but mayybe change to `ScoreSlimChunks` or something? 

I changed from it OnlyId

File: server/src/handlers/chunk_handler.rs (hunk 1 above)
Comment (skeptrunedev) [inline server/src/handlers/chunk_handler.rs:1105]:
```suggestion
            ("application/json.slim_chunks" = SearchChunkQuerySlimChunksResponseBody)
```

File: server/src/handlers/chunk_handler.rs [hunk 3]
```diff
@@ -1294,7 +1302,11 @@ pub struct RecommendChunksRequest {
     tag = "chunk",
     request_body(content = RecommendChunksRequest, description = "JSON request payload to get recommendations of chunks similar to the chunks in the request", content_type = "application/json"),
     responses(
-        (status = 200, description = "JSON response payload containing chunks with scores which are similar to those in the request body", body = Vec<ChunkMetadataWithFileData>),
+
+        (status = 200, content(
+            ("application/json" = Vec<ChunkMetadata>),
+            ("application/json.only_ids" = Vec<SlimChunkMetadata>)
```
Comment (skeptrunedev) [inline server/src/handlers/chunk_handler.rs:1308]: Same `slim_chunks` comment

**Review by skeptrunedev — CHANGES_REQUESTED** [review]

**Comment by fastfists** [comment]
Looks a lot better!

**Comment by skeptrunedev** [comment]
image

**Review by fastfists — APPROVED** [review]

File: server/src/lib.rs [hunk 4]
```diff
@@ -286,6 +291,12 @@ impl Modify for SecurityAddon {
             data::models::DatasetUsageCount,
             data::models::ClientDatasetConfiguration,
             data::models::StripePlan,
+            data::models::SearchChunkQueryIDsResponseBody,
```
Comment (skeptrunedev) [inline server/src/lib.rs:294]: `SlimChunks`
```suggestion
            data::models::SearchSlimChunksQueryResponseBody,
```

File: server/src/lib.rs [hunk 5]
```diff
@@ -286,6 +291,12 @@ impl Modify for SecurityAddon {
...
+            data::models::ScoreIDs,
```
Comment (skeptrunedev) [inline server/src/lib.rs:295]:
```suggestion
            data::models::ScoreSlimChunks
```

File: server/src/lib.rs [hunk 6]
```diff
@@ -286,6 +291,12 @@ impl Modify for SecurityAddon {
...
+            data::models::SlimChunkMetadata,
+            data::models::SearchGroupIDsResult,
```
Comment (skeptrunedev) [inline server/src/lib.rs:297]:
```suggestion
            data::models::SearchGroupSlimChunksResult,
```

File: server/src/lib.rs [hunk 7]
```diff
@@ -286,6 +291,12 @@ impl Modify for SecurityAddon {
...
+            data::models::SearchOverGroupsIDsResponseBody,
```
Comment (skeptrunedev) [inline server/src/lib.rs:298]:
```suggestion
            data::models::SearchOverGroupsSlimChunksResponseBody,
```

File: server/src/lib.rs [hunk 8]
```diff
@@ -286,6 +291,12 @@ impl Modify for SecurityAddon {
...
+            data::models::GroupIDsDTO,
```
Comment (skeptrunedev) [inline server/src/lib.rs:299]:
```suggestion
            data::models::GroupSlimChunksDTO,
```

**Review by skeptrunedev — CHANGES_REQUESTED** [review]

**Review by skeptrunedev — APPROVED** [review]

**Pull request merged**